larray/core/axis.py

```diff
@@ -234,6 +236,60 @@ def by(self, length, step=None, template=None):
         """
         return self[:].by(length, step, template)
 
+    def astype(self, dtype: Union[str, np.dtype], casting: str = 'unsafe', inplace: bool = False) -> 'Axis':
+        """
+        Cast labels to a specified type.
+
+        Parameters
+        ----------
+        dtype: str or dtype
+            Typecode or data-type to which the labels are cast.
+
+        casting: str, optional
+            Controls what kind of data casting may occur. Defaults to ‘unsafe’.
+
+                * ‘no’ means the data types should not be cast at all.
+                * ‘equiv’ means only byte-order changes are allowed.
+                * ‘safe’ means only casts which can preserve values are allowed.
+                * ‘same_kind’ means only safe casts or casts within a kind, like float64 to float32, are allowed.
+                * ‘unsafe’ means any data conversions may be done.
+
+        inplace: bool, optional
```

Comment:
I **WONDER** whether or not we should implement any inplace operation in larray. For now, we implement only very very few of them and I feel like we should either implement that in as many as make sense or not at all. I agree that this one is more likely to be useful but well...